src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,112 @@
+
+public class Duke {
+
+    Scanner sc;
+    Ui ui;
+    Parser parser;
+    Storage storage;
+    ArrayList<Task> tasks;
+    TaskList taskList;
+
+    public Duke() {
+        sc = new Scanner(System.in);
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage("data/duke.txt");
+        tasks = new ArrayList<>();
+    }
+
+    public String run () {
+        String status;
+        try {
+            tasks = storage.printFileContents();
+            status = "Successfully retrieved tasks";
+        } catch (FileNotFoundException e) {
+            status = "File not found";
+        }
+        taskList = new TaskList(tasks);
+        return status;
+    }
+
+    /**
+     * Return the response message based on the input command from the user.
+     *
+     * @param input the input command from the user.
+     * @return the response message
+     */
+    protected String getResponse(String input) {
+        String responseMessage;
+        String command = parser.parseCommand(input);
+        if (command.equals("list")) {
+            taskList.listTask();
+            responseMessage = taskList.taskMessage;
+        } else if (command.equals("done")) {
+            int index = Integer.parseInt(input.split(" ")[1]) - 1;
+            if (index < tasks.size() && index >= 0) {
+                taskList.markTaskDone(index);
+                storage.writeToFile(tasks);
+                responseMessage = taskList.taskMessage;
+            } else {
+                responseMessage = ui.taskNotExist();
+            }
+        } else if (command.equals("delete")) {
+            int index = Integer.parseInt(input.split(" ")[1]) - 1;
+            try {
+                taskList.deleteTask(index);
+                storage.writeToFile(tasks);
+                responseMessage = taskList.taskMessage;
+            } catch (Exception e) {
+                responseMessage = ui.taskNotExist();
+            }
+        } else if (command.equals("todo")) {
+            int emptyIndex = input.indexOf(" ");
+            if (input.trim().equals("todo")) {
+                responseMessage = ui.emptyDescription();
+            } else {
+                taskList.addToDo(input.substring(emptyIndex + 1));
+                storage.writeToFile(tasks);
+                responseMessage = taskList.taskMessage;
+            }
+        } else if (command.equals("deadline")) {
+            int emptyIndex = input.indexOf(" ");
+            int slashIndex = input.indexOf("/");
+            if (input.trim().equals("deadline")) {
+                responseMessage = ui.emptyDescription();
+            } else if (slashIndex == -1) {
+                responseMessage = ui.emptyDate();
+            } else {
+                taskList.addDeadline(input.substring(emptyIndex + 1, slashIndex - 1), LocalDate.parse(input.substring(slashIndex + 4)));
+                storage.writeToFile(tasks);
+                responseMessage = taskList.taskMessage;
```

Comment:
Consider moving all the evaluation of user input into Parser class as well, and have the parseInput(), or whatever the method will be called, return a Deadline object